python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/router-bgp.j2

```diff
@@ -946,6 +946,291 @@ router bgp {{ router_bgp.as }}
+      neighbor {{ neighbor.ip_address }} next-hop resolution v4-mapped-v6 translation
+{%             elif neighbor.next_hop_resolution.v4_mapped_v6_translation is arista.avd.defined(false) %}
+      neighbor {{ neighbor.ip_address }} next-hop resolution v4-mapped-v6 translation disabled
+{%             endif %}
+{%         endfor %}
+{%         if router_bgp.address_family_ipv4_labeled_unicast.networks is arista.avd.defined %}
+{%             for network in router_bgp.address_family_ipv4_labeled_unicast.networks %}
+{%                 set network_cli = "network " ~ network.prefix %}
+{%                 if network.route_map is arista.avd.defined %}
+{%                     set network_cli = network_cli ~ " route-map " ~ network.route_map %}
+{%                 endif %}
+      {{ network_cli }}
+{%             endfor %}
+{%         endif %}
+{%         if router_bgp.address_family_ipv4_labeled_unicast.next_hops is arista.avd.defined %}
+{%             for next_hop in router_bgp.address_family_ipv4_labeled_unicast.next_hops %}
+{%                 set next_hop_cli = "next-hop " ~ next_hop.ip_address ~ " originate" %}
+{%                 if next_hop.lfib_backup_ip_forwarding is arista.avd.defined(true) %}
+{%                     set next_hop_cli = next_hop_cli ~ " lfib-backup ip-forwarding" %}
+{%                 endif %}
+      {{ next_hop_cli }}
+{%             endfor %}
+{%         endif %}
+{%         if router_bgp.address_family_ipv4_labeled_unicast.lfib_entry_installation_skipped is arista.avd.defined(true) %}
+      lfib entry installation skipped
+{%         endif %}
+{%         if router_bgp.address_family_ipv4_labeled_unicast.label_local_termination is arista.avd.defined %}
+      label local-termination {{ router_bgp.address_family_ipv4_labeled_unicast.label_local_termination }}
+{%         endif %}
+{%         if router_bgp.address_family_ipv4_labeled_unicast.graceful_restart is arista.avd.defined(true) %}
+      graceful-restart
+{%         endif %}
+{%         if router_bgp.address_family_ipv4_labeled_unicast.neighbor_default.next_hop_self is arista.avd.defined(true) %}
+      neighbor default next-hop-self
+{%         endif %}
+{%         if router_bgp.address_family_ipv4_labeled_unicast.next_hop_resolution_ribs is arista.avd.defined %}
+{%             set v4_bgp_lu_resolution_ribs = [] %}
+{%             for rib in router_bgp.address_family_ipv4_labeled_unicast.next_hop_resolution_ribs %}
+{%                 if rib.rib_type is arista.avd.defined("tunnel-rib-colored") %}
+{%                     do v4_bgp_lu_resolution_ribs.append("tunnel-rib colored system-colored-tunnel-rib") %}
+{%                 elif rib.rib_type is arista.avd.defined("tunnel-rib") and rib.rib_name is arista.avd.defined() %}
+{%                     do v4_bgp_lu_resolution_ribs.append("tunnel-rib " ~ rib.rib_name) %}
+{%                 elif rib.rib_type is arista.avd.defined %}
```

Comment:
So if a user defined rib.rib_type as `tunnel-rib` but did not defined the rib.rib_name then this condition still valid and we will append only `tunnel-rib` in v4_bgp_lu_resolution_ribs which is not correct.

Reply:
Right, we need the middle elif case to be a nested one. First check for `tunnel-rib` and then an inner if to check for rib_name. So we just ignore the tunnel-rib if rib_name is missing.

Colin just took what we had already under some other address family, so this issue should also be fixed there.

@MaheshGSLAB please add suggestions to make it easy to just commit them.